lib/azure/armrest/configuration.rb

```diff
@@ -93,8 +96,12 @@ def initialize(args)
           :grant_type   => 'client_credentials',
           :proxy        => ENV['http_proxy'],
           :ssl_version  => 'TLSv1',
+          :max_threads  => 10
         }.merge(args.symbolize_keys)
 
+        # Avoid thread safety issues for VCR testing.
+        options[:max_threads] = 1 if defined?(VCR)
+
```

Comment:
@djberg96 - If we do not set ```VCR``` how will we ever set the max_threads to 1?
Im trying to understand the logic, its not that i do not like it.